### ISSUE TITLE
Allow `var` inside `declare namespace` on no-var-keyword rule

### ISSUE THREAD

**Issue by saschanaz** [palantir/tslint#2796]
### Bug Report

- __TSLint version__: tslint@next (5.0.0-dev.0)
- __TypeScript version__: master as of 21 May 2017
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
declare namespace Intl {
    interface CollatorOptions {
        usage?: string;
        localeMatcher?: string;
        numeric?: boolean;
        caseFirst?: string;
        sensitivity?: string;
        ignorePunctuation?: boolean;
    }

    interface ResolvedCollatorOptions {
        locale: string;
        usage: string;
        sensitivity: string;
        ignorePunctuation: boolean;
        collation: string;
        caseFirst: string;
        numeric: boolean;
    }

    interface Collator {
        compare(x: string, y: string): number;
        resolvedOptions(): ResolvedCollatorOptions;
    }
    var Collator: {
        new (locales?: string | string[], options?: CollatorOptions): Collator;
        (locales?: string | string[], options?: CollatorOptions): Collator;
        supportedLocalesOf(locales: string | string[], options?: CollatorOptions): string[];
    };
}
```

with `tslint.json` configuration:

```json
{
    "rules": {
        "no-var-keyword": true
    }
}
```

#### Actual behavior

[25, 5]: Forbidden 'var' keyword, use 'let' or 'const' instead

#### Expected behavior

`var` inside `declare namespace` should be allowed as `declare var` is

**Comment by ajafff** [comment]
Why can't you use `let` or `const`? Does this cause any problems in real world code?

**Comment by SaschaNaz** [comment]
lib.d.ts uses this, currently the file is excluded on CI for linting, and it seems the only blocker there is this issue. (Another was JSDoc related one which can be easily fixed by text editors)

**Comment by adidahiya** [comment]
> lib.d.ts uses this

should `tsc` be generating `let` instead then?

either way, I think it'd be fine to change the rule to allow `var` inside `declare namespace` for `.d.ts` files only

**Comment by SaschaNaz** [comment]
>should tsc be generating let instead then?

Then the `prefer-const` rule complains. It seems it's a bug too but I couldn't make a minimal repro for it yet.

**Comment by ajafff** [comment]
Newer versions of `prefer-const` ignore declaration files and other ambient declarations

**Comment by SaschaNaz** [comment]
@ajafff Oh, so it was a recent change. I tried reinstalling tslint@next (which is currently TS depends on) but couldn't get the latest changed bit. I'm not sure as I can get it without `@next` thing, is `tslint@next` not supported now?

**Comment by ajafff** [comment]
@next wasn't updated since tslint 5.0.0
tslint@latest allows typescript@next as peer dependency, hence there would be no version mismatch if typescript switched to tslint@latest

**Comment by adidahiya** [comment]
Good point, I didn't realize typescript was still on next. Filed https://github.com/Microsoft/TypeScript/issues/16039

**Comment by JoshuaKGoldberg** [comment]
☠️ *TSLint's time has come!* ☠️ 

TSLint is no longer accepting most feature requests per #4534. See **[typescript-eslint.io](https://typescript-eslint.io)** for the new, shiny way to lint your TypeScript code with **ESLint**. ✨ 

It was a pleasure open sourcing with you all!
